aspnet/client-side/bundling-and-minification.rst

```diff
@@ -1,8 +1,301 @@
+
+.. code-block:: javascript
+
+	<environment names="Development">
+		<link rel="stylesheet" href="~/lib/bootstrap/dist/css/bootstrap.css" />
+		<link rel="stylesheet" href="~/lib/bootstrap-touch-carousel/dist/css/bootstrap-touch-carousel.css" />
+		<link rel="stylesheet" href="~/css/site.css" />
+	</environment>
+	<environment names="Staging,Production">
+		<link rel="stylesheet" href="//ajax.aspnetcdn.com/ajax/bootstrap/3.0.0/css/bootstrap.min.css"
+				asp-fallback-href="~/lib/bootstrap/css/bootstrap.min.css"
+				asp-fallback-test-class="hidden" asp-fallback-test-property="visibility" asp-fallback-test-value="hidden" />
+		<link rel="stylesheet" href="//ajax.aspnetcdn.com/ajax/bootstrap-touch-carousel/0.8.0/css/bootstrap-touch-carousel.css"
+				asp-fallback-href="~/lib/bootstrap-touch-carousel/css/bootstrap-touch-carousel.css"
+				asp-fallback-test-class="carousel-caption" asp-fallback-test-property="display" asp-fallback-test-value="none" />
+		<link rel="stylesheet" href="~/css/site.min.css" asp-file-version="true" />
+	</environment>
+	
+Switching Between Environments
+------------------------------
+
+To switch between compiling for different environments, you can change the ASPNET_ENV environment variable. 
+1.	In **Task Runner Explorer**, verify that the **min** task has been set to occur **Before Build**.
+2.	In **Solution Explorer**, right-click the project name and select **Properties**.
+	The property sheet for the Web app is displayed.
+3.	Set the value of the **ASPNET_ENV** environment variable to ``Production``.
+4.	Press **F5** to run the application in a browser.
+5.	In the browser window, right-click the page and select **View Source** to see the html for the page.
+	You will notice that the stylesheet links point to the minified CSS files.
+6.	Close the browser to stop the Web app.
+7.	In Visual Studio, return to the property sheet for the Web app and change the **ASPNET_ENV** environment variable back to ``Development``.
+8.	Press **F5** to run the application in a browser again.
+9.	In the browser window, right-click the page and select **View Source** to see the html for the page.
+	You will notice that the stylesheet links point to the full version of the CSS files.
+	
+	For more information related to Visual Studio 2015 environments, see `Working with Multiple Environments <http://docs.asp.net/en/latest/fundamentals/environments.html>`_.
+	
+Task and Module Details
+-----------------------
+A gulp task is registered with a function name.  You can specify dependencies if other tasks must run before the current task. Additional functions allow you to run and watch the gulp tasks, as well as set the source (src) and destination (dest) of the files that you are modifying. The following are the primary gulp functions:
+
+•	gulp.task(name[, deps], fn) { }
+•	gulp.run(tasks) { }
```

Comment:
Make this a table 0 first column what you have, 2nd column what it does - also explain glob, globs,